### PR TITLE
Variable tilt clamp

### PR DESCRIPTION
**Pull request by adityaruplaha**
Follow up of #3154 .
Allows to set the tilt clamp to a value other than 90 degrees. I'm no good with the UI, so if this gets merged, someone else will have create a UI for this.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/3366)

### CODE REVIEW

**Comment by adityaruplaha** [comment]
How do I set the default value to 90 degrees? Sorry bu I'm not very familiar with the Citra settings implementation.

**Comment by adityaruplaha** [comment]
The only problem I'm having is setting the tilt clamp value correctly without introducing a circular dependency (again). Maybe my previous commit will do the trick. Or it won't currently, I am seeing the Appveyor bield proceed.

**Comment by adityaruplaha** [comment]
Huh? How come my new setting doesn't show up in the config settings? Can anyone explain what's happening? As I said, I'm not very good with this `settings` stuff

**Comment by wwylele** [comment]
You need to make a user interface to set `Values::tilt_clamp`, or at least expose it in the ini file, otherwise the variable remains uninitialized. The code should be placed along with other settings in the frontend ([here](https://github.com/citra-emu/citra/blob/master/src/citra/config.cpp) and [here](https://github.com/citra-emu/citra/blob/master/src/citra_qt/configuration/config.cpp)). 

Also, as a convention of the current code, [all settings for `MotionEmu` is read out from its `ParamPackage`](https://github.com/citra-emu/citra/blob/9b647d459bbb2ac45f821273e8d3a85ab12b60e6/src/input_common/motion_emu.cpp#L141-L142), and these settings do not become members of `core/settings`, but become members of `MotionEmuDevice`, so you may want to follow the same rule.

Did you follow my suggestion to test the game beforehand?

**Comment by adityaruplaha** [comment]
I can't test it, coz I don't have the game, but I'm pretty sure that it doesn't work, having received feedback from some other people. But how do I expose this setting?

**Comment by wwylele** [comment]
So here is the report from discord:


>wwylele / 白疾風 - Today at 2:06 PM
>right click mouse on the screen and drag it upward to the maximum when it levels up to 30
could anyone try this now?
>
>Mike© - Today at 2:13 PM
>Let me try
>
>
>Mike© - Today at 2:33 PM
>@wwylele / 白疾風 it evolved
>

So it is already doable with current implementation.

**Comment by adityaruplaha** [comment]
Hmmmm, did my friends pull off a joke on me? 💢 
Anyways, I'll still go ahead with, coz this might help people, some day (maybe).

**Comment by adityaruplaha** [comment]
Um, can anyone tell me how to format my code, so that clang-format succeeds?

**Comment by BreadFish64** [comment]
You should be able to find a clang-format plugin for visual studio

**Comment by BreadFish64** [comment]
You should be able to find a clang-format plugin for visual studio

**Comment by BreadFish64** [comment]
You should be able to find a clang-format plugin for visual studio

**Comment by adityaruplaha** [comment]
I use CodeBlocks.

**Comment by BreadFish64** [comment]
Then you can apply the diff given by Travis

**Comment by adityaruplaha** [comment]
I don't understand exactly what it means by "trailing whitespace". Sorry, if I'm acting stupid 😞 .

**Comment by adityaruplaha** [comment]
No reson for this to fail right now. Please add `pr:needs-review` & also `pr:needs-work` if you think that's required.

**Comment by adityaruplaha** [comment]
Did as much as I can, please review.

File: src/input_common/motion_emu.cpp [hunk 1]
```diff
@@ -11,17 +11,18 @@
...
 // Implementation class of the motion emulation device
 class MotionEmuDevice {
 public:
-    MotionEmuDevice(int update_millisecond, float sensitivity)
+    MotionEmuDevice(int update_millisecond, float sensitivity, float tilt_clamp)
         : update_millisecond(update_millisecond),
           update_duration(std::chrono::duration_cast<std::chrono::steady_clock::duration>(
               std::chrono::milliseconds(update_millisecond))),
-          sensitivity(sensitivity), motion_emu_thread(&MotionEmuDevice::MotionEmuThread, this) {}
+          sensitivity(sensitivity), motion_emu_thread(&MotionEmuDevice::MotionEmuThread, this) { this->tilt_clamp = tilt_clamp;}
```
Comment (wwylele) [inline src/input_common/motion_emu.cpp]: You might want to follow the style how `sensitivity` and `update_duration` is initialized above. You can write `tilt_clamp(tilt_clamp)` in the initializer list (before `{}`)

File: src/input_common/motion_emu.cpp [hunk 2]
```diff
@@ -44,7 +45,7 @@ class MotionEmuDevice {
             } else {
                 tilt_direction = mouse_move.Cast<float>();
                 tilt_angle = MathUtil::Clamp(tilt_direction.Normalize() * sensitivity, 0.0f,
-                                             MathUtil::PI * 0.5f);
+                                             MathUtil::PI * (this->tilt_clamp)/180.0f);
```
Comment (wwylele) [inline src/input_common/motion_emu.cpp]: No need for `()`

Comment (adityaruplaha) [reply src/input_common/motion_emu.cpp]: Ya, but enhances readability, so i thought... 
Does it cause an clang-format error? I'm worried sick abt how to remove them.

File: src/input_common/motion_emu.cpp (hunk 1 above)
Comment (adityaruplaha) [reply src/input_common/motion_emu.cpp]: I dont know how to do that. Frankly, its the first time i'm seeing stuff being done like that.

**Comment by adityaruplaha** [comment]
Please add `pr:needs-review`

**Comment by adityaruplaha** [comment]
Ummmmm, I wish I could, but this is the first time I'm using clang-format, so I don't understand anything that diff says. Specially this part
image
Can you please help me out here @wwylele ?

**Comment by adityaruplaha** [comment]
Fixed everything!

**Comment by adityaruplaha** [comment]
Can confirm that clang-format & MinGW succeeded. 😃

**Comment by adityaruplaha** [comment]
Yippee! My first contribution to citra merged!

TODO:
Implement a UI for this feature.
Any UI coder here? I'm no good at that.

**Pull request merged**